src/Context/Todo/TodoContainer.jsx

```diff
@@ -1,18 +1,36 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import { uuid } from "../../helpers/uuid";
 import { TodoProvider } from "./TodoContext";
 import { todoStoreDefaults } from "./TodoDefaults";
+import { addTodo, getTodos } from "./TodoServices";
 
 export default function TodoContainer({ children }) {
   const [state, setState] = useState(todoStoreDefaults);
 
-  const createTodo = (newTodoValue) => {
-    setState((prevState) => {
-      return {
-        ...prevState,
-        todoList: [...prevState.todoList, { name: newTodoValue, id: uuid() }],
-      };
-    });
+  async function load() {
+    const data = await getTodos();
+    setState((prev) => ({
+      ...prev,
+      todoList: [...data],
+    }));
+  }
+
+  async function add(newTodo) {
+    const data = await addTodo(newTodo);
+  }
+
+  useEffect(() => {
+    try {
+      load();
+    } catch (err) {
+      console.log(err);
+    }
+  }, [add]);
+
+  const createTodo = (newTodoValue, callback) => {
+    console.log(newTodoValue);
+    add(newTodoValue);
```

Comment:
you need to add await here else it will not stop and call **callback()** immediately.
a callback is a function we want to run after we complete the required operation we need to perform (in this case, it is the creation of todo).